snuba/query/processors/project_rate_limiter.py

```diff
@@ -0,0 +1,104 @@
+from snuba.query.processors import QueryProcessor
+from snuba.query.logical import Query
+from snuba.query.conditions import (
+    ConditionFunctions,
+    get_first_level_and_conditions,
+)
+from snuba.query.expressions import FunctionCall, Literal
+from snuba.query.matchers import Any as AnyMatch
+from snuba.query.matchers import Column as ColumnMatch
+from snuba.query.matchers import FunctionCall as FunctionCallMatch
+from snuba.query.matchers import Literal as LiteralMatch
+from snuba.query.matchers import String as StringMatch
+from snuba.query.matchers import Or, Param
+from snuba.request.request_settings import RequestSettings
+from snuba.state import get_configs
+from snuba.state.rate_limit import PROJECT_RATE_LIMIT_NAME, RateLimitParameters
+
+
+class ProjectRateLimiterProcessor(QueryProcessor):
+    """
+    If there isn't already a rate limiter on a project, search the top level conditions
+    for project IDs using the given project column name and add a rate limiter for them.
```

Comment:
According to PEP8 we should keep comment and dosctring lines within 72 chars
https://www.python.org/dev/peps/pep-0008/#:~:text=Limit%20all%20lines%20to%20a,be%20limited%20to%2072%20characters.

This helps readability when doing code reviews making long docstrings readable without lines being cut.